### PR TITLE
Add Basic SongSelectScreen Elements

### PR DESCRIPTION
**Pull request by S1lentzer**

### CODE REVIEW

File: Rhythmic/Other/Beatmap/BeatmapMetadata.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
```
Comment (AltenGD) [inline Rhythmic/Other/Beatmap/BeatmapMetadata.cs]: This file should be deleted if its not getting used

File: Rhythmic/Other/CarouselItem.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Audio.Sample;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Configuration;
+using osu.Framework.Input.Events;
+
+
+namespace Rhythmic.Other
+{
+    public enum CarouselBeatmapState
```
Comment (AltenGD) [inline Rhythmic/Other/CarouselItem.cs]: Should be at the bottom instead at the top

File: Rhythmic/Other/CarouselItem.cs [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        Load,
+    }
+    class CarouselItem : Container
+    {
+        private Container borderContainer;
+        private Container nestedContainer;
+        private Box hoverLayer;
+        public readonly Bindable<CarouselBeatmapState> State = new Bindable<CarouselBeatmapState>(CarouselBeatmapState.UnSelected);
+
+
```
Comment (AltenGD) [inline Rhythmic/Other/CarouselItem.cs]: Remove useless spaces

File: Rhythmic/Other/CarouselItem.cs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                borderContainer.BorderThickness = 0;
+                borderContainer.EdgeEffect = new EdgeEffectParameters
+                {
+                    Type = EdgeEffectType.Shadow,
+                    Offset = new Vector2(1),
+                    Radius = 10,
+                    Colour = Color4.Black.Opacity(100),
+                };
+                State.Value = CarouselBeatmapState.UnSelected;
+                //State.Value = CarouselBeatmapState.Load; Use only in DrawableCarouselItem, to load the specified beatmap
```
Comment (AltenGD) [inline Rhythmic/Other/CarouselItem.cs]: If any code is commented, please delete them

File: Rhythmic/Other/RhytmicBoxObject.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
```
Comment (AltenGD) [inline Rhythmic/Other/RhytmicBoxObject.cs]: Again, if its not used, delete the file

File: Rhythmic/Other/Beatmap/BeatmapMetadata.cs (hunk 1 above)
Comment (S1lentzer) [reply Rhythmic/Other/Beatmap/BeatmapMetadata.cs]: fixed

File: Rhythmic/Other/CarouselItem.cs (hunk 2 above)
Comment (S1lentzer) [reply Rhythmic/Other/CarouselItem.cs]: fixed

**Pull request merged**